diag_manager/diag_manager.F90

```diff
@@ -4561,6 +4558,46 @@ SUBROUTINE diag_field_add_attribute_i1d(diag_field_id, att_name, att_value)
     endif
   END SUBROUTINE diag_field_add_attribute_i1d
 
+  !> @brief Add a scalr attribute to the diag field corresponding to a given id
+  subroutine diag_field_add_attribute_0d(diag_field_id, att_name, att_value)
+    INTEGER,          INTENT(in) :: diag_field_id !< ID number for field to add attribute to
+    CHARACTER(len=*), INTENT(in) :: att_name      !< new attribute name
+    class(*),         INTENT(in) :: att_value     !< new attribute value
+
+    select type(att_value)
+    type is (real(kind=r4_kind))
+      call diag_field_add_attribute_scalar_r(diag_field_id, att_name, real(att_value))
+    type is (real(kind=r8_kind))
+      call diag_field_add_attribute_scalar_r(diag_field_id, att_name, real(att_value))
+    type is (integer(kind=i4_kind))
```

Comment:
In the old diag manager, it ends up always writing the att_value as NF90_FLOAT. 
https://github.com/NOAA-GFDL/FMS/blob/d79c3b53c04e81ba009533aa3f6428d102868432/diag_manager/diag_manager.F90#L4547
In https://github.com/NOAA-GFDL/FMS/pull/1625/commits/4829fe88739cde63cbb349320b020062b3f74225, I made the changes so that the new diag manager uses the actual precision of att_value.